### PR TITLE
added edge cases for SerializeKeyParms and refactored test

### PR DESCRIPTION
**Pull request by gh4683**
- Added checks to make sure there are no RSA key params for AES algorithms and no Symmetric key params for RSA algorithms
- Refactored test to use toBytes() function and separated success and failure cases
- Added test cases for the edge cases added

### CODE REVIEW

File: service/biz/tpm12_utils.go
[the remark this answers is not in the document]
Comment (gh4683) [reply service/biz/tpm12_utils.go:1]: sure done

File: service/biz/tpm12_utils.go [hunk 1]
```diff
@@ -950,13 +950,19 @@ func (u *DefaultTPM12Utils) SerializeKeyParms(keyParms *TPMKeyParms) ([]byte, er
 	var err error
 	switch keyParms.AlgID {
 	case tpm12.AlgRSA:
+		if keyParms.Params.SymParams != nil {
+			return nil, fmt.Errorf("unexpected symmetric key params for algorithm %v", keyParms.AlgID)
```
[the remark this answers is not in the document]
Comment (gh4683) [reply service/biz/tpm12_utils.go]: yes makes sense. fixed

File: service/biz/tpm12_utils_test.go [hunk 2]
```diff
@@ -1139,6 +1127,77 @@ func TestSerializeKeyParms(t *testing.T) {
...
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := u.SerializeKeyParms(tc.keyParms)
+			if err == nil {
+				t.Fatalf("SerializeKeyParms(%+v) got nil error, want error containing %q", tc.keyParms, tc.expectedError)
+			}
+			if !strings.Contains(err.Error(), tc.expectedError) {
+				t.Errorf("SerializeKeyParms(%+v) got error %v, want error containing %q", tc.keyParms, err, tc.expectedError)
```
[the remark this answers is not in the document]
Comment (gh4683) [reply service/biz/tpm12_utils_test.go]: i see, that makes sense, thanks for sharing! I am using errors.Is now and we can change tests as we modify them to use errors.Is going forward

**Review by gscert — APPROVED** [review]
